cmd/apichecker/apicheck_test.go

```diff
@@ -0,0 +1,692 @@
+		}
+	}
+	if updatedExch == exchCheck {
+		t.Fatal("update failed")
+	}
+}
+
+func TestCheckMissingExchanges(t *testing.T) {
+	t.Parallel()
+	a := checkMissingExchanges()
+	if len(a) > len(exchange.Exchanges) {
+		t.Fatal("invalid response")
+	}
+}
+
+func TestNameUpdates(t *testing.T) {
+	t.Parallel()
+	tester := []struct {
+		Name          string
+		Status        string
+		ExpectedName  string
+		ErrorExpected bool
+	}{
+		{
+			Name:          "incorrectname",
+			Status:        "incomplete",
+			ErrorExpected: true,
+		},
+		{
+			Name:          "Gemini 2 2",
+			Status:        "incomplete",
+			ErrorExpected: false,
+		},
+		{
+			Name:          " Gemini 23",
+			Status:        "incomplete",
+			ErrorExpected: true,
+		},
+		{
+			Name:          "Gemini 123",
+			Status:        "complete",
+			ExpectedName:  "Gemini 1",
+			ErrorExpected: false,
+		},
+		{
+			Name:          "Gemini",
+			Status:        "complete",
+			ExpectedName:  "Gemini 1",
+			ErrorExpected: false,
+		},
+		{
+			Name:          "Gemini 24 ",
+			Status:        "incomplete",
+			ErrorExpected: false,
+		},
+	}
+	for x := range tester {
+		r, err := nameStateChanges(tester[x].Name, tester[x].Status)
+		if r != tester[x].ExpectedName && err != nil && !tester[x].ErrorExpected {
+			t.Errorf("%d failed, expected %v, %v, got: %v, %v\n", x,
+				tester[x].ExpectedName,
+				tester[x].ErrorExpected,
+				r,
+				err)
+		}
+	}
+}
+
+func TestReadFileData(t *testing.T) {
+	t.Parallel()
+	_, err := readFileData(testJSONFile)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetSha(t *testing.T) {
+	t.Parallel()
+	_, err := getSha("binance-exchange/binance-official-api-docs")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCheckBoardID(t *testing.T) {
+	if !areAPIKeysSet() {
+		t.Skip()
+	}
+	a, err := trelloCheckBoardID()
+	if err != nil {
+		t.Error(err)
+	}
+	if a != true {
+		t.Error("no match found for the given boardID")
+	}
+}
+
+func TestTrelloGetLists(t *testing.T) {
+	if !areAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := trelloGetLists()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetAllCards(t *testing.T) {
+	if !areAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := trelloGetAllCards()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetAllChecklists(t *testing.T) {
+	if !areAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := trelloGetAllChecklists()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestTrelloGetAllBoards(t *testing.T) {
+	if !areAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := trelloGetBoardID()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCreateNewList(t *testing.T) {
+	if !areAPIKeysSet() {
+		t.Skip()
+	}
+	err := trelloCreateNewList()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestTrelloCreateNewCard(t *testing.T) {
+	if !areAPIKeysSet() {
+		t.Skip()
+	}
+	err := trelloCreateNewCard()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCreateNewChecklist(t *testing.T) {
+	if !areAPIKeysSet() {
```

Comment:
Please change to `!canTestTrello()`

Reply:
canTestTrello checks if all the things r set like boardid AND cardid AND listID AND ...
whereas to test a singular function such as testGetAllLists we only need boardid, the rest can be blank, so i check if apikeys r set n then in the function i check if trelloboard id != ""

Reply:
The test `TestTrelloGetAllBoards` fails because of the way these tests have been setup. It is reading from `updates.json` when no testing api key fields have been set. This allows tests to move past `!areAPIKeysSet()` even when no keys have been set inside the tests. This applies to all the tests I've tagged. 
So the application grabs the api keys from the `updates.json` file, which shouldn't happen and then `TestTrelloGetAllBoards` fails because because `trelloBoardName` is empty. You shouldn't be using keys from the `updates.json` file in the same way that when running tests in gocryptotrader, we don't use our live api keys from `config.json`

Reply:
I'm okay if you just check for the fields you need inside each test like you say, but `areAPIKeysSet` is not good for these tests